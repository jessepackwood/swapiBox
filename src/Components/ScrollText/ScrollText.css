.scroll-container {
  background-size: cover;
  background-position: center;
  color: #feda4a;
  display: flex;
  font-family: 'Pathway Gothic One', sans-serif;
  height: 200px;
  justify-content: center;
  letter-spacing: 6px;
  overflow: hidden;
  perspective: 350px;
  position: relative;
  text-align: justify; }

.scroll-container .scroll-text {
  animation: scroll-text 175s linear;
  font-size: 35px;
  position: relative;
  text-align: center;
  top: -100px;
  transform-origin: 0% 0%;
  width: 80%; }

.scroll-container .scroll-text h3 {
  margin: 0px; }

.scroll-container .scroll-text h5 {
  margin: 10px 0px; }

@keyframes scroll-text {
  0% {
    top: 500px;
    transform: rotateX(20deg) translateZ(0); }
  100% {
    top: -5000px;
    transform: rotateX(25deg) translateZ(-2500px); } }
