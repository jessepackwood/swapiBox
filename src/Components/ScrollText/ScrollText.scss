.scroll-container {
  background-size: cover;
  background-position: center;
  color: #feda4a;
  display: flex;
  font-family: 'Pathway Gothic One', sans-serif;
  height: 500px;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
  justify-content: center;
  letter-spacing: 6px;
  overflow: hidden;
  perspective: 350px;
  position: relative;
  text-align: justify;
}

.scroll-container .scroll-text {
  animation: scroll-text 120s linear;
  font-size: 40px;
  position: relative;
  text-align: center;
  top: -100px;
  transform-origin: 0% 0%;
  width: 80%; }

  .scroll-container .scroll-text h3 {
    margin: 0px; 
  }

  .scroll-container .scroll-text h5 {
    margin: 10px 0px; 
  }



@keyframes scroll-text {
  0% {
    top: 500px;
    transform: rotateX(20deg) translateZ(0); }
  100% {
    top: -5000px;
    transform: rotateX(25deg) translateZ(-2500px); } }