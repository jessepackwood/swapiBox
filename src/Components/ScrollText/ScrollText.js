import React from 'react';
import './ScrollText.css';
import Button from '../Button/Button';
import PropTypes from 'prop-types';

class ScrollText extends React.Component {
  
  componentDidMount() {
    const elements = document.getElementsByClassName('scroll-text');
    elements[0].addEventListener('animationend', this.props.onSkip );
  }

  render() {
    const {film, onSkip} = this.props;
    const romanNumerial = {  
      1: 'I',
      2: 'II',
      3: 'III',
      4: 'IV',
      5: 'V',
      6: 'VI',
      7: 'VII'
    };

    return (
      <div>
        <div className="scroll-container">
          <div className="scroll-text">
            <h3>{film.title}</h3>
            <h5>EPISODE {romanNumerial[film.episode]}</h5>
            <h4>{film.text}</h4>
          </div>
        </div>
        <Button
          className='btn-skip'
          buttonText='Skip' 
          functionToFire={onSkip}
        />
      </div>
    );
  }
}

export default ScrollText;

ScrollText.propTypes = {
  film: PropTypes.object,
  onSkip: PropTypes.func
};