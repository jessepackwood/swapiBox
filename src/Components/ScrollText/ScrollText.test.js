import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import ScrollText from './ScrollText.js'

describe('ScrollText tests', () => {
  beforeEach() => {
    const scrollText = shallow(<ScrollText />)
  }
  });