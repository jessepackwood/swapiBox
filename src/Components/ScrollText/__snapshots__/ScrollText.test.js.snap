// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollText
    film={
        Object {
            "episode": 1,
            "text": "Scrolling",
            "title": "Star Wars",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="scroll-container"
>
          <div
                    className="scroll-text"
          >
                    <h3>
                              Star Wars
                    </h3>
                    <h5>
                              EPISODE 
                              I
                    </h5>
                    <h4>
                              Scrolling
                    </h4>
          </div>
</div>,
        <Button
          buttonText="Skip"
          className="btn-skip"
          functionToFire={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="scroll-text"
>
            <h3>
                        Star Wars
            </h3>
            <h5>
                        EPISODE 
                        I
            </h5>
            <h4>
                        Scrolling
            </h4>
</div>,
          "className": "scroll-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Star Wars
</h3>,
              <h5>
                EPISODE 
                I
</h5>,
              <h4>
                Scrolling
</h4>,
            ],
            "className": "scroll-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Star Wars",
              },
              "ref": null,
              "rendered": "Star Wars",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "EPISODE ",
                  "I",
                ],
              },
              "ref": null,
              "rendered": Array [
                "EPISODE ",
                "I",
              ],
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Scrolling",
              },
              "ref": null,
              "rendered": "Scrolling",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "Skip",
          "className": "btn-skip",
          "functionToFire": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="scroll-container"
>
            <div
                        className="scroll-text"
            >
                        <h3>
                                    Star Wars
                        </h3>
                        <h5>
                                    EPISODE 
                                    I
                        </h5>
                        <h4>
                                    Scrolling
                        </h4>
            </div>
</div>,
          <Button
            buttonText="Skip"
            className="btn-skip"
            functionToFire={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="scroll-text"
>
              <h3>
                            Star Wars
              </h3>
              <h5>
                            EPISODE 
                            I
              </h5>
              <h4>
                            Scrolling
              </h4>
</div>,
            "className": "scroll-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Star Wars
</h3>,
                <h5>
                  EPISODE 
                  I
</h5>,
                <h4>
                  Scrolling
</h4>,
              ],
              "className": "scroll-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Star Wars",
                },
                "ref": null,
                "rendered": "Star Wars",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "EPISODE ",
                    "I",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "EPISODE ",
                  "I",
                ],
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Scrolling",
                },
                "ref": null,
                "rendered": "Scrolling",
                "type": "h4",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Skip",
            "className": "btn-skip",
            "functionToFire": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
